spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -727,51 +740,97 @@ def write_sorting(sorting, save_path, **nwbfile_kwargs):
                     spkt = sorting.get_unit_spike_train(unit_id=id) / fs # spike trains withinin the SortingExtractor object are not scaled by sampling frequency
                     nwbfile.add_unit(id=id, spike_times=spkt)
 
+            # Always add column for shank ids if a MultiSortingExtractor
+            if isinstance(sorting,se.MultiSortingExtractor):
+                set_dynamic_table_property(
+                        dynamic_table=nwbfile.units,
+                        row_ids=[int(k) for k, v in sorting._unit_map.items()],
+                        property_name='shank_id',
+                        values=[unit_dict['unit_id']-1 for ind,unit_dict in sorting._unit_map.items()],
+                        default_value=np.nan,
+                        description='0-indexed id of cluster of shank'
+                )
+
             # Units properties
             for pr in all_properties:
-                unit_ids = [int(k) for k, v in sorting._properties.items()
+                unit_ids = [int(k) for k, v in total_prop_dict.items()
                             if pr in v]
-                vals = [v[pr] for k, v in sorting._properties.items()
-                        if pr in v]
+                
+                # We can always guarantee two levelx of nested dictionaries based
```

Comment:
This is unclear.. what do you mean?

Reply:
Sorry, it should read

# We can always guarantee two levels of nested dictionaries
# based on the current structure

which refers to the fact that if there is a unit property, it was added via the basic set_unit_property, which by line 331 of `SortingExtractor`

`self._properties[unit_id][property_name] = value`

implies there are two levels to the dictionary such that 

`unit_prop = next(iter(next(iter(total_prop_dict.values())).values()))`

will not return an error by trying to iterate or call values over a non-dictionary object.